### PR TITLE
Add a flag to disable read_database

### PR DESCRIPTION
**Pull request by ardichoke**
I'm currently in the process of deploying standalone puppetdb hosts, which connect to a managed database on a cloud provider. The entire read_database section caused me a large amount of unnecessary confusion and are simply unnecessary when not running a hot standby cluster. This adds a simple flag to turn read_database config off. Defaults the setting to true in params in order to not change default functionality.

### CODE REVIEW

**Review by bastelfreak — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -562,7 +562,7 @@ The name of the PuppetDB service. Defaults to `puppetdb`.
 
 #### `puppetdb_service_status`
 
-Sets whether the service should be `running ` or `stopped`. When set to `stopped` the
+Sets whether the service should be `running` or `stopped`. When set to `stopped` the
```
Comment (bastelfreak) [inline README.md:565]: nitpick: such README.md cleanups could also be submitted in a dedicated PR if we aim for perfect/clean PRs. But I think it's also fine to keep the changes here.

Comment (ardichoke) [reply README.md:565]: Yeah, that was a byproduct of my editor's auto-cleanup tool. :facepalm:

**Comment by ardichoke** [comment]
Don't want to be a nag, but this has been approved since February, can this get merged?

**Comment by ardichoke** [comment]
Anyone? Bueller?

**Comment by smortex** [comment]
Why has the test suite not run against this PR?  Closing and re-opening to see if it helps…

**Comment by smortex** [comment]
Does not seems to help.  Can you please rebase your changes on top of the main branch?

From your branch in your fork of the repo:

```
git fetch origin       # Download the latest code we have here
git rebase origin/main # Move your commits on top of the main branch
git push -f            # Send the changes (-f is required because we re-wrote history)
```

**Comment by ardichoke** [comment]
Rebased... as requested. Still doesn't seem to be running the test suite. I have to imagine because this PR predates the test suite but no one ever bothered merging it.

**Comment by h0tw1r3** [comment]
@ardichoke recently the test suite was overhauled. Can you rebase please? I expect there will be some work to fix the conflicts and update the spec tests. If you need assistance with updating the tests, happy to help.

**Comment by ardichoke** [comment]
I'm dropping this.... it has been 2 years since I authored this simple change and no one has bothered to merge it. I'm not wasting even more of my personal time sorting out 2 years worth of merge conflicts when it will likely languish even longer thanks to the neglect of Perforce.
